src/main/java/com/spotify/futures/CompletableFutures.java

```diff
@@ -423,6 +423,44 @@ public static void checkCompleted(CompletionStage<?> stage) {
                                       ff.join()));
   }
 
+  /**
+   * Combines multiple stages by applying a function.
+   *
+   * @param function the combining function.
+   * @param stages   the stages to combine
+   * @param <T>      the type of the combining function's return value.
+   * @return a stage that completes into the return value of the supplied function.
+   * @since 0.4.0
+   */
+  public static <T> CompletionStage<T> combine(
+          Function<Combined, T> function, CompletionStage<?>... stages) {
```

Comment:
Here I feel like we could have one overload delegate to the other via `Arrays.asList()`

Reply:
Yes, we could. Should we prefer the code reduction/deduplication over potential performance benefits?

Reply:
Yes, give me a benchmark if you want to prove me wrong :)